src/main/java/org/folio/rest/impl/FinanceHelper.java

```diff
@@ -0,0 +1,205 @@
+
+  FinanceHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(httpClient, okapiHeaders, ctx, lang);
+  }
+
+  /**
+   * Creates Encumbrance records associated with given PO line and updates PO line with corresponding links.
+   *
+   * @return CompletableFuture with void on success.
+   */
+  CompletableFuture<Void> handleEncumbrances(CompositePurchaseOrder compPo) {
+    List<Pair<Transaction, FundDistribution>> encumbranceDistributionPairs = buildEncumbrances(compPo);
+    List<Transaction> encumbrances = encumbranceDistributionPairs.stream()
+      .map(Pair::getLeft)
+      .collect(toList());
+    return setCurrentFiscalYear(encumbrances)
+      .thenCompose(aVoid -> createSummary(compPo.getId(), encumbrances.size()))
+      .thenCompose(vVoid -> createEncumbrances(encumbranceDistributionPairs));
+  }
+
+  private CompletableFuture<String> createSummary(String id, int number) {
+    if (number < 1) {
+      return CompletableFuture.completedFuture(id);
+    }
+    OrderTransactionSummary summary = new OrderTransactionSummary()
+      .withId(id)
+      .withNumTransactions(number);
+    return createRecordInStorage(JsonObject.mapFrom(summary), resourcesPath(ORDER_TRANSACTION_SUMMARIES));
+  }
+
+  private CompletableFuture<Void> createEncumbrances(List<Pair<Transaction, FundDistribution>> encumbranceDistributionPairs) {
+    return VertxCompletableFuture.allOf(ctx, encumbranceDistributionPairs.stream()
+      .map(pair -> createRecordInStorage(JsonObject.mapFrom(pair.getLeft()), String.format(ENCUMBRANCE_POST_ENDPOINT, lang))
+        .thenAccept(id -> pair.getValue()
+          .setEncumbrance(id)))
+      .toArray(CompletableFuture[]::new));
+  }
+
+  private CompletableFuture<Void> setCurrentFiscalYear(List<Transaction> encumbrances) {
+    Map<String, List<Transaction>> groupedByFund = encumbrances.stream()
+      .collect(groupingBy(Transaction::getFromFundId));
+    return groupByLedgerIds(groupedByFund).thenCompose(groupedByLedgerId -> VertxCompletableFuture.allOf(ctx, groupedByLedgerId.entrySet()
+      .stream()
+      .map(entry -> getCurrentFiscalYear(entry.getKey()).thenAccept(fiscalYear -> entry.getValue()
+        .forEach(transaction -> transaction.setFiscalYearId(fiscalYear.getId()))))
+      .collect(toList())
+      .toArray(new CompletableFuture[0])));
+  }
+
+  private CompletableFuture<Map<String, List<Transaction>>> groupByLedgerIds(Map<String, List<Transaction>> groupedByFund) {
+    return collectResultsOnSuccess(StreamEx.ofSubLists(new ArrayList<>(groupedByFund.entrySet()), MAX_IDS_FOR_GET_RQ)
+      .map(entries -> entries.stream()
+        .map(Map.Entry::getKey)
+        .distinct()
+        .collect(toList()))
+      .map(this::getFundsByIds)
+      .toList()).thenApply(
+          lists -> lists.stream()
+            .flatMap(Collection::stream)
+            .collect(HashMap::new, accumulator(groupedByFund), Map::putAll));
+  }
+
+  @NotNull
```

Comment:
Is this annotation necessary here?

Reply:
No. Maybe it was generated by IDE during refactoring